src/Http/Authentication.Core/src/AuthenticationSchemeProvider.cs

```diff
@@ -47,14 +47,19 @@ protected AuthenticationSchemeProvider(IOptions<AuthenticationOptions> options,
 
     private readonly IDictionary<string, AuthenticationScheme> _schemes;
     private readonly List<AuthenticationScheme> _requestHandlers;
+    private AuthenticationScheme? _autoDefaultScheme;
```

Comment:
Is it possible to store scheme with the Task in order to avoid creating it each time?
```suggestion
    private Task<AuthenticationScheme?> _autoDefaultScheme;
```

Reply:
Sure done